diff --git a/src/demo/player.rs b/src/demo/player.rs
index 3bd404e..b1a95ed 100644
--- a/src/demo/player.rs
+++ b/src/demo/player.rs
@@ -20,7 +20,8 @@ pub(super) fn plugin(app: &mut App) {
     // Record directional input as movement controls.
     app.add_systems(
         Update,
-        record_player_directional_input
+        // add the new sprint system to the Update schedule
+        (record_player_directional_input, record_sprint)
             .in_set(AppSystems::RecordInput)
             .in_set(PausableSystems),
     );
@@ -91,6 +92,25 @@ fn record_player_directional_input(
     }
 }
 
+fn record_sprint(
+    input: Res<ButtonInput<KeyCode>>,
+    mut controller_query: Query<&mut MovementController, With<Player>>,
+) {
+    // define the normal speed for the player. This is defined 2 times, here and in the `MovementController` default function. This is not good practice but we leave it for now, because that means refactoring which we don't want at this point. (we could write the default values into some serialized form as a file or define a global player `Resource`)
+    let normal_speed = 400.;
+
+    // get the movement controller. We only have one so this is only executed once.
+    for mut controller in &mut controller_query {
+        let mut speed = normal_speed;
+        // change to the sprint speed when holding down shift to three times the normal value.
+        if input.pressed(KeyCode::ShiftLeft) {
+            speed *= 3.0;
+        };
+        // apply max speed change to the controller.
+        controller.max_speed = speed;
+    }
+}
+
 #[derive(Resource, Asset, Clone, Reflect)]
 #[reflect(Resource)]
 pub struct PlayerAssets {
