diff --git a/src/demo/movement.rs b/src/demo/movement.rs
index aa91b91..4574dfd 100644
--- a/src/demo/movement.rs
+++ b/src/demo/movement.rs
@@ -46,7 +46,7 @@ impl Default for MovementController {
         Self {
             intent: Vec2::ZERO,
             // 400 pixels per second is a nice default, but we can still vary this per character.
-            max_speed: 400.0,
+            max_speed: 2000.0,
         }
     }
 }
